src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversion.cs

```diff
@@ -67,6 +67,18 @@ internal bool IsArrayIndex
             }
         }
 
+        private class DeconstructionUncommonData : UncommonData
```

Comment:
I may keep that for a follow-up. My main goal was to not clutter UncommonData, but I didn't want to engage in any deep refactorings there.